lsp-mode.el

```diff
@@ -4512,21 +4512,20 @@ If INCLUDE-DECLARATION is non-nil, request the server to include declarations."
   was most recently invoked.")
 
 (defun lsp--document-highlight ()
-  (unless (or (looking-at "[[:space:]\n]")
-              (not lsp-enable-symbol-highlighting)
-              (and (lsp--point-on-highlight?)
-                   (let ((curr-sym-bounds (bounds-of-thing-at-point 'symbol)))
-                     (prog1
-                         (and curr-sym-bounds
-                              (equal curr-sym-bounds
-                                     lsp--symbol-bounds-of-last-highlight-invocation))
-                       (setq lsp--symbol-bounds-of-last-highlight-invocation
-                             curr-sym-bounds)))))
-    (lsp-request-async "textDocument/documentHighlight"
-                       (lsp--text-document-position-params)
-                       #'lsp--document-highlight-callback
-                       :mode 'tick
-                       :cancel-token :highlights)))
+  (let ((curr-sym-bounds (bounds-of-thing-at-point 'symbol)))
+    (unless (or (looking-at "[[:space:]\n]")
+                (not lsp-enable-symbol-highlighting)
+                (and (lsp--point-on-highlight?)
```

Comment:
lsp--point-on-highlight? is redundant now because practically it will be always overridden by current symbol check.

Reply:
OK, I removed that invocation.

Reply:
(and the function itself).

Reply:
> (and the function itself).

I am not quite sure about that though. there is one more usage.

Reply:
I'd prefer to keep the function. I think it would be messier if we replaced its invocation with a `bounds-of-thing-at-point` check of some kind.

Reply:
Yes. Actually I like the current implementation very much. It removes the overlays when you go out of the overlay and it refreshes the overlay when you move out of the current symbol.